### PR TITLE
GRAD2-2017: monthly & yearly & supplemental distribution jobs are ref…

### PR DESCRIPTION
**Pull request by infstar**
…actored to be async non-blocking IO.

GRAD2-2017: monthly & yearly & supplemental distribution jobs are refactored to be async non-blocking IO.

### CODE REVIEW

File: api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java [hunk 1]
```diff
@@ -90,59 +90,59 @@ public class DistributionServiceTest {
...
 	@Test
 	public void testdistributeCredentialsMonthly() {
-		DistributionResponse res = testdistributeCredentials_transcript("MER","USERDIST",false,null);
+		DistributionResponse res = testdistributeCredentials_transcript("MER","MONTHLYDIST",false,null, true);
 		assertNotNull(res);
-		res = testdistributeCredentials_certificate("MER","USERDIST","YED2",null,true);
+		res = testdistributeCredentials_certificate("MER","MONTHLYDIST","YED2",null,true, true);
 		assertNotNull(res);
-		res = testdistributeCredentials_certificate("MER","USERDIST","YEDB",null,false);
+		res = testdistributeCredentials_certificate("MER","MONTHLYDIST","YEDB",null,false, true);
```
Comment (cditcher) [inline api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java:97]: Getting unit test failure here (line 97)

File: api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java [hunk 2]
```diff
@@ -90,59 +90,59 @@ public class DistributionServiceTest {
...
 		assertNotNull(res);
 	}
 
 	@Test
 	public void testdistributeCredentialsYearly() {
-		DistributionResponse res = testdistributeCredentials_transcript("MERYER","USERDIST",false,null);
+		DistributionResponse res = testdistributeCredentials_transcript("MERYER","DISTRUN_YE",false,null, true);
 		assertNotNull(res);
-		res = testdistributeCredentials_certificate("MERYER","USERDIST","YED2",null,false);
+		res = testdistributeCredentials_certificate("MERYER","DISTRUN_YE","YED2",null,false, true);
```
Comment (cditcher) [inline api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java:141]: Getting unit test failure here (line 141)

**Review by cditcher — CHANGES_REQUESTED** [review]

File: api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java (hunk 2 above)
Comment (infstar) [reply api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java:141]: The activity code, DISTRUN_YE is right for Year-end dist run.   The wrong one was used.

File: api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java (hunk 1 above)
Comment (infstar) [reply api/src/test/java/ca/bc/gov/educ/api/distribution/service/DistributionServiceTest.java:97]: This test is for monthly dist run.  So the activity code, MONTHLYDIST is right.

**Review by cditcher — APPROVED** [review]

**Pull request merged**